helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java

```diff
@@ -451,9 +474,11 @@ public static boolean siblingNodesActiveReplicaCheck(HelixDataAccessor dataAcces
         if (stateByInstanceMap.containsKey(instanceName)) {
           int numHealthySiblings = 0;
           for (Map.Entry<String, String> entry : stateByInstanceMap.entrySet()) {
-            if (!entry.getKey().equals(instanceName) && (toBeStoppedInstances == null
-                || !toBeStoppedInstances.contains(entry.getKey())) && !unhealthyStates.contains(
-                entry.getValue())) {
+            String siblingInstanceName = entry.getKey();
+            if (!siblingInstanceName.equals(instanceName) && (toBeStoppedInstances == null
+                || !toBeStoppedInstances.contains(siblingInstanceName))
+                && !unhealthyStates.contains(entry.getValue()) && !isOperationSetForInstance(
+                dataAccessor, siblingInstanceName)) {
```

Comment:
This increase the dimension of checks. Why not just let the instances already in operation in toBeStoppedInstances? Semantically it makes sense as well.

Reply:
Sounds good. Will delegate the evacuating node filtering to `StoppablInstancesSelector`, in which it would go through all instances in the topology and add evacuating nodes to the `toBeStoppedInstances`